search/util.go

```diff
@@ -137,10 +137,15 @@ type GeoBufferPoolCallbackFunc func() *s2.GeoBufferPool
 
 const KnnPreSearchDataKey = "_knn_pre_search_data_key"
 const SynonymPreSearchDataKey = "_synonym_pre_search_data_key"
+const BM25PreSearchDataKey = "_bm25_pre_search_data_key"
 
 const PreSearchKey = "_presearch_key"
 
-type ScoreExplCorrectionCallbackFunc func(queryMatch *DocumentMatch, knnMatch *DocumentMatch) (float64, *Explanation)
+const SearchTypeKey = "_search_type_key"
+const FetchStatsAndSearch = "fetch_stats_and_search"
+
+const SearcherStartCallbackKey = "_searcher_start_callback_key"
+const SearcherEndCallbackKey = "_searcher_end_callback_key"
```

Comment:
I realize this is older code, but just to resolve some tech-debt as well here - would you add some commentary to what each key here stands for.